Django-Start-Project.md

```diff
@@ -0,0 +1,265 @@
+# Creating a Project with Django
+
+[Home](Python)
+
+Now that we know how to create virtual environments and use pip, we can begin
+building our project. In this article, we will start our first Django project,
+create tests, and start our development server.
+
+### Creating the Virtual Environment
+
+First, let's create a new virtual environment for this project. (If you haven't
+already, deactivate the previous virtualenv by typing `deactivate` in the
+terminal). Navigate to the directory where you want
+the Django project and type the following into the terminal:
+```
+mkvirtualenv taskplanner --python=/usr/bin/python3
+```
+
+You may have to change your Python path if it looks different than the one
+above.
+
+The command line shell should now have `(taskplanner)` before the rest of the
+text. 
+
+If it doesn't look like that, just type:
+```
+workon taskplanner
+```
+
+We can now install Django:
+```
+pip install Django
+```
+
+### Create our Django Project
+
+With Django installed we can create our project:
+```
+django-admin.py startproject taskplanner
+```
+
+Next, navigate into our new project by typing:
+```
+cd taskplanner
+```
+
+Before we do anything, let's set this directory as our working directory for
+our virtual environment:
+```
+setvirtualenvproject
+```
+
+Now, we can simply type `cdproject` in our terminal to bring us back to this directory from
+anywhere.
+
+Your project directory should look something like this:
+```
+taskplanner (our main working directory)
+|--- manage.py (similar to the django-admin script, you will see this used a
+lot throughout our project)
+|--- taskplanner
+    |--- __init__.py (this just tells python to treat this directory as a package)
+    |--- settings.py (main configuration file for our project)
+    |--- urls.py (we will use this to configure urls)
+    |--- wsgi.py (this is used for deploying our project to a production server)
+```
+
+### Functional Testing
+
+Test-driven development is a widely used best practice in developing software.
+Bascially, we want to write a test first that is bound to fail, and then write
+the least amount of code possible to pass that test. With Django, our goal is to write
+both functional tests, and unit tests throughout development. Don't sweat it,
+testing isn't as difficult as it seems.
```

Comment:
Sounds a bit patronizing. Might want to soften the blow with an exclamation mark!